artifactory.py

```diff
@@ -2084,6 +2082,163 @@ def symlink_to(self, target, target_is_directory=False):
         raise NotImplementedError()
 
 
+class ArtifactoryBuild(ArtifactoryPath):
+    def __new__(cls, *args, **kwargs):
+        """
+        pathlib.Path overrides __new__ in order to create objects
+        of different classes based on platform. This magic prevents
+        us from adding an 'auth' argument to the constructor.
+        So we have to first construct ArtifactoryPath by Pathlib and
+        only then add auth information.
+        """
+        obj = super().__new__(cls, *args, **kwargs)
+        obj.project = kwargs.get("project", "")
+        return obj
+
+    @property
+    def builds(self):
+        """
+        Get all available builds on Artifactory
+        :return: (list) list of available build names
+        """
+        all_builds = []
+        url = ""
+        if self.project:
+            url = f"?project='{self.project}'"
+
+        obj = self.joinpath(url)
+        resp = self._accessor.get_response(obj).json()
+
+        if "builds" in self._get_build_api_response(url):
+            for build in resp["builds"]:
+                all_builds.append(build["uri"][1:])
+
+        return all_builds
```

Comment:
Is `all_builds`'s type `str`? 

Look weird - we create an instance of `ArtifactotyBuild`, then get `ArtifactotyBuild.builds` and we get `List[str]`...
In this case `ArtifactoryBuild` actually is `ArtifactotyBuildManager`

Reply:
Do you propose to rename class or change method behaviour?

Reply:
Yep, I left more a comment with more information https://github.com/devopshq/artifactory/pull/248/files#r667658652

I'll close this converstion